File: core/src-test/com/oni/donjon/generator/ImproveRectangleTest.java
package com.oni.donjon.generator;

import com.badlogic.gdx.math.Rectangle;
import org.junit.Assert;
import org.junit.Test;

/**
 * @author Daniel Chesters (on 21/04/17).
 */
public class ImproveRectangleTest {


    @Test
    public void testIntersectsTrue() {
        Rectangle rectangle1 = new Rectangle(1, 1, 3, 3);
        Rectangle rectangle2 = new Rectangle(2, 2, 3, 3);
        Assert.assertTrue(
            com.oni.donjon.generator.ImproveRectangleKt.intersects(rectangle1, rectangle2));
    }

    @Test
    public void testIntersectsFalse() {
        Rectangle rectangle1 = new Rectangle(10, 10, 1, 1);
        Rectangle rectangle2 = new Rectangle(5, 5, 3, 3);
        Assert.assertFalse(com.oni.donjon.generator.ImproveRectangleKt.intersects(rectangle1, rectangle2));
    }
}

File: core/src/test/kotlin/com/oni/donjon/generator/ImproveRectangleTest.kt
package com.oni.donjon.generator

import com.badlogic.gdx.math.Rectangle
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

/**
 * @author Daniel Chesters (on 21/04/17).
 */
class ImproveRectangleTest {

    @Test
    fun testIntersectsTrue() {
        val rectangle1 = Rectangle(1f, 1f, 3f, 3f)
        val rectangle2 = Rectangle(2f, 2f, 3f, 3f)
        assertTrue(rectangle1.intersects(rectangle2))
    }

    @Test
    fun testIntersectsFalse() {
        val rectangle1 = Rectangle(10f, 10f, 1f, 1f)
        val rectangle2 = Rectangle(5f, 5f, 3f, 3f)
        assertFalse(rectangle1.intersects(rectangle2))
    }
}
